### PR TITLE
New AppMetr Android

### PR DESCRIPTION
**Pull request by ZorVladimir**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+}
+
+dependencies {
+    provided 'com.google.android.gms:play-services-ads:10.0.1'
+}
+
+task jar(type: Jar, dependsOn: 'assembleRelease') {
+    from fileTree(dir: 'build/intermediates/classes/release')
+}
```
Comment (chemist777) [inline app/build.gradle]: Не хватает пустой строки в конце файла.

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 2]
```diff
@@ -7,12 +7,15 @@
...
 import android.content.pm.PackageManager;
+import android.os.AsyncTask;
 import android.os.Build;
 import android.os.Handler;
 import android.util.Log;
 import com.appmetr.android.AppMetrListener;
 import com.appmetr.android.BuildConfig;
 import com.appmetr.android.internal.command.CommandsManager;
+import com.google.android.gms.ads.identifier.AdvertisingIdClient;
+
```
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Why we need this empty line?

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 3]
```diff
@@ -685,4 +695,33 @@ protected boolean verifyPaymentAndCheck(String purchaseInfo, String receipt, Str
...
+     */
+    private class GetGAIDTask extends AsyncTask<Context, Integer, String> {
+        @Override
+        protected String doInBackground(Context... contexts)
+        {
+            try {
+                AdvertisingIdClient.Info info = AdvertisingIdClient.getAdvertisingIdInfo(contexts[0]);
+
+                if(info.isLimitAdTrackingEnabled()) {
+                    return "";
```
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: If it ok that we return an empty line here but check result on equality to `null`?

Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: If we need to return an empty line here I would suggest adding comment why.

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java (hunk 2 above)
Comment (ZorVladimir) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: We don't. I used it to divide imports into external and internal

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java (hunk 3 above)
Comment (ZorVladimir) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Empty line means that we can't get GAID, null means that we are still in progress of getting GAID

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java (hunk 2 above)
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: I would suggest having the same code style through all sources within project. As far as I can see with have code style where blank lines is between `java.*` imports and all other imports. So would ask you to remove this empty line here.

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java (hunk 3 above)
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Understood. My question was more about is it intended that we emit an empty line as `mobGoogleAid` to AppMetr backend? We might have a lot of empty `mobGoogleAid` values and querying by empty line might not makes sense. Maybe we can just filter them on client side but it's better to ask guys from analytic team.

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 4]
```diff
@@ -585,6 +591,10 @@ private void removeBatchFile(String fileName) {
         ret.add(new BasicNameValuePair("mobLibVer", LibraryPreferences.VERSION_STRING));
         ret.add(new BasicNameValuePair("mobAndroidID", mRequestParameters.ANDROID_ID));
 
+        if(mGoogleAID != null) {
```
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Could you please add space after `if`

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 5]
```diff
@@ -685,4 +695,33 @@ protected boolean verifyPaymentAndCheck(String purchaseInfo, String receipt, Str
...
+     * Sets GoogleAID
+     */
+    private class GetGAIDTask extends AsyncTask<Context, Integer, String> {
+        @Override
+        protected String doInBackground(Context... contexts)
+        {
+            try {
+                AdvertisingIdClient.Info info = AdvertisingIdClient.getAdvertisingIdInfo(contexts[0]);
+
+                if(info.isLimitAdTrackingEnabled()) {
```
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Could you please add space after `if`

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 6]
```diff
@@ -685,4 +695,33 @@ protected boolean verifyPaymentAndCheck(String purchaseInfo, String receipt, Str
...
+                    return "";
+                } else {
+                    return info.getId();
+                }
+            } catch (final Throwable t) {
+                if(BuildConfig.DEBUG) {
```
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Could you please add space after `if`

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 7]
```diff
@@ -685,4 +695,33 @@ protected boolean verifyPaymentAndCheck(String purchaseInfo, String receipt, Str
...
         }
     }
+
+    /**
+     * Sets GoogleAID
+     */
+    private class GetGAIDTask extends AsyncTask<Context, Integer, String> {
+        @Override
+        protected String doInBackground(Context... contexts)
+        {
```
Comment (logarithm) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: `{` should be on the previous line.

File: app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java (hunk 3 above)
Comment (ZorVladimir) [inline app/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: On a server, we'll see that mobGoogleAid was sent as "", so some error happened. And a server will not add it to parameters of a player. They have a filter

**Comment by ZorVladimir** [comment]
https://github.com/appmetr/appmetr-android/pull/2

**Pull request merged**